### DISCUSSION TITLE
Child themes (👍12)

### DISCUSSION THREAD

**Discussion by Ryuno-Ki** [11ty/eleventy#4021] [Enhancement Queue]
I recently moved over from harp.js to eleventy. What I love on the architecture here is the [layout chaining](https://www.11ty.io/docs/layouts/#layout-chaining).

But what about theming? And more: Allowing for [child themes](https://developer.wordpress.org/themes/advanced-topics/child-themes/).

Note, that it is also [coming up in Jekyll](https://github.com/jekyll/jekyll/issues/7344).

I'd expect this to be shipped in a eleventy v2.x earliest. Before we can think about themes, we should have a first stable version out there and then make it more smooth.

*What would it allow you?*

It would let you focus on getting the content done and not worry too much about how it will look like.
The community could provide themes which you can select by setting it in your`.eleventy.js` config. Need inspiration?

* [hexo themes](https://hexo.io/themes/)
* [hugo themes](https://themes.gohugo.io/)
* [Jekyll Themes](http://jekyllthemes.org/)
* [Jekyll Themes](https://jekyllthemes.io/)

*What's needed to realise this?*

A standard API on how to bundle layout files with plugins (because most themes will be made up of those). Some common language on how to expose variables (I think, we can use a data file here).

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by paulrobertlloyd** [comment]
I’d love to see this. I was trying to hack something together using `eleventyConfig.addLayoutAlias` in my plugin, but that didn’t work:

`Error: You’re trying to use a layout that does not exist: post (undefined)`

Also unclear how paths would be expected to work in this case – relative to plugin or relative to the site in which plugin is being used?

Another option might be to allow `dir.layouts`/`dir.includes` to accept an array of values… and to be able to set/append paths via the Plugin API.

Just mentioning some ideas out loud. But yes; it’d make managing and distributing themes so much easier if this could be achieved via plugins.
